rfc/rfc-0000-allow-resources-to-report-status.md

```diff
@@ -0,0 +1,294 @@
+3. `multiMatch`
+    * If defined, matchers for both the 'Healthy' state and the 'Unhealthy' state must be defined.
+    * The matchers in the 'Healthy' state are ANDed together, whereas the matchers in the 'Unhealthy' state are ORed.
+      * That is, a resource is considered to be 'Healthy' only if _all of_ the matchers are fulfilled, but is considered to be 'Unhealthy' if _any one of_ the matchers are fulfilled.
+    * The supported matchers are `matchConditions` and `matchFields`. Each is an array, and an author can define either or both of these matchers for a given state.
+    * For `matchConditions`, authors need only define the `type` and `status` of the condition for Cartographer to read. In the owner status, the full condition will be reflected (with the `reason`, `message`, etc.)
+    * For `matchFields`, the spec is based off of `matchFields` in blueprint resource options. Authors need to define the `key` and `operator`, and potentially the `values` (if the operator is `In` or `NotIn`). Authors can also optionally define a `messagePath`, which is a path in the resource where Cartographer can pull off a message to display in the owner status.
+    
+Each owner's status.resources will now include a `conditions` field. To make room for future extensibility, this will be an array of conditions, though this RFC only defines one condition to be included here. The condition will be of `type` 'Healthy' and will report a `status` of 'True', 'False', or 'Unknown'. The condition will also have a `reason` which will be based off the `healthyConditionRule` as follows:
+    
+* If `alwaysHealthy: {}`, `reason: AlwaysHealthy`.
+* If `singleConditionType: X`, `reason: XCondition`.
+* If `multiMatch: ...`, `reason: (MatchedCondition|MatchedField)`.
+
+If the `healthyConditionRule` is not `alwaysHealthy`, the condition will also have a message. For `singleConditionType`, the message will be propagated from the specified condition. For `multiMatch`, the message will specify the value of the matched condition or field, and then display the message of the matched condition or the message at the path specified in `matchFields`.
+
+In the case of `multiMatch`, if there is more than one matching condition or field that caused the resource to enter it's current state, the first one read will be propagated by Cartographer.
+
+Each owner will display a new condition in the status called `ResourcesHealthy` which will only be updated to `True` when all resources in the blueprint are healthy. Cartographer will not update the top level state of an owner to `Ready: True` until `ResourcesHealthy` is `True`.
+
+# Migration
+[migration]: #migration
+
+Given that Cartographer will default to `Healthy` once the resource is applied and outputs are read, this will not be a breaking change.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+- We only allow for reflecting three states, though this design allows for creating more states in the future.
```

Comment:
clarify the use of `states` here, it could either be the conditions (of which we define one) or the status of each condition which will always be `True`, `False`, or `Unknown`